.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,79 @@
+name: Pre-commit checks
+
+on: [pull_request, push]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: ${{ contains(github.event.pull_request.labels.*.name, 'pre-commit open PR') && '0' || '100' }}
```

Comment:
Labels are only available to people with triage (?) permissions.

This could also be triggered with a comment instead:

https://github.com/opengisch/QField/blob/57863640a6067331d31cb011c34431b05ac80ac3/.github/workflows/pre-commit.yml#L36-L43

Reply:
good point, fixed